data/vector.go

```diff
@@ -447,6 +447,58 @@ func NewFieldFromFieldType(p FieldType, n int) *Field {
 	return f
 }
 
+// NewNullableFieldFromFieldType creates a new nullable Field of the given pType of length n.
+func NewNullableFieldFromFieldType(p FieldType, n int) *Field {
+	f := &Field{}
+	switch p {
+	// ints
+	case FieldTypeInt8, FieldTypeNullableInt8:
+		f.vector = newNullableInt8Vector(n)
+
+	case FieldTypeInt16, FieldTypeNullableInt16:
+		f.vector = newNullableInt16Vector(n)
+
+	case FieldTypeInt32, FieldTypeNullableInt32:
+		f.vector = newNullableInt32Vector(n)
+
+	case FieldTypeInt64, FieldTypeNullableInt64:
+		f.vector = newNullableInt64Vector(n)
+
+	// uints
+	case FieldTypeUint8, FieldTypeNullableUint8:
+		f.vector = newNullableUint8Vector(n)
+
+	case FieldTypeUint16, FieldTypeNullableUint16:
+		f.vector = newNullableUint16Vector(n)
+
+	case FieldTypeUint32, FieldTypeNullableUint32:
+		f.vector = newNullableUint32Vector(n)
+
+	case FieldTypeUint64, FieldTypeNullableUint64:
+		f.vector = newNullableUint64Vector(n)
+
+	// floats
+	case FieldTypeFloat32, FieldTypeNullableFloat32:
+		f.vector = newNullableFloat32Vector(n)
+
+	case FieldTypeFloat64, FieldTypeNullableFloat64:
+		f.vector = newNullableFloat64Vector(n)
+
+	// other
+	case FieldTypeString, FieldTypeNullableString:
+		f.vector = newNullableStringVector(n)
+
+	case FieldTypeBool, FieldTypeNullableBool:
+		f.vector = newNullableBoolVector(n)
+
+	case FieldTypeTime, FieldTypeNullableTime:
+		f.vector = newNullableTimeTimeVector(n)
+	default:
+		panic(fmt.Sprint("unsupported vector ptype"))
```

Comment:
I think it would be helpful for anyone debugging, to include the field type in the error (is there a way to get the name of the type instead?):
```suggestion
		panic(fmt.Sprintf("unsupported vector ptype: %+v", p))
```

Reply:
`NewNullableFieldFromFieldType` I think might be confusing, since NewFieldFromFieldType can create nullablefields as well.  FieldType already has a `Nullable()` bool method. But maybe FieldType.NullableType() which always returns the nullable type (like this).